evm_arithmetization/src/prover.rs

```diff
@@ -523,20 +523,20 @@ fn build_segment_data<F: RichField>(
     }
 }
 
-pub struct SegmentDataIterator {
+pub struct SegmentDataIterator<'a> {
     pub partial_next_data: Option<GenerationSegmentData>,
-    pub inputs: GenerationInputs,
+    pub inputs: &'a GenerationInputs,
     pub max_cpu_len_log: Option<usize>,
```

Comment:
Isn't this redundant? `max_cpu_len_log` is already contained in the `GenerationSegmentData`